File: chapter1/python/test1.py
import cv2 as cv
from matplotlib import pyplot as plt
"""
cv.IMREAD_COLOR 1 显示三色图
cv.IMREAD_GRAYSCALE 0 显示灰度图
cv.IMREAD_UNCHANGED 加载图像，包括alpha通道
"""
img = cv.imread("../../images/ambush_5_left.jpg", 1)

b, g, r = cv.split(img)
new_image = cv.merge([r, g, b])

cv.namedWindow("image", cv.WINDOW_AUTOSIZE)
cv.imshow('image-bgr', img)
cv.imshow("image-rgb", new_image)

# plt.imshow(img)
# plt.show()

k = cv.waitKey(0)

if k == ord('q'):
    # save image in annother place
    cv.imwrite("./temp.jpg", img)
    cv.destroyAllWindows()
elif k == 27:
    cv.destroyAllWindows()
    

File: chapter1/cpp/test1.cpp
#include<opencv2/opencv.hpp>
#include<iostream>
#include<string>
#include "opencv2/core/hal/interface.h"
#include <filesystem>
using namespace std;
using namespace cv;

int main(){
    //这里好像不能使用相对路径
    string imagePath = "/home/chengyingjie666/code/learnOpencv/images/230501201_1.jpg";
    cv::Mat img = cv::imread(imagePath, 1);

    // 打印文件路径
    cout << "Image path: " << imagePath << endl;

    // 检查文件是否存在
    if (!filesystem::exists(imagePath)) {
        cerr << "Error: File does not exist at path " << imagePath << endl;
        return -1;
    }

    if (img.empty()) {
        cerr << "Error: Could not open or find the image" << endl;
        return -1;
    }
    // swap channles and assign new order channels into a new mat,and imshow both mat.
    vector<cv::Mat> channels;
    cv::split(img, channels);
    cv::Mat new_image;
    cv::swap(channels[0], channels[1]);
    cv::merge(channels, new_image);

    cv::imshow("image-bgr", img);
    cv::imshow("new-image-rgb", new_image);
    cv::waitKey(0);
    return 0;
}

